common/web/gesture-recognizer/src/tools/unit-test-resources/src/inputSequenceSimulator.ts

```diff
@@ -0,0 +1,315 @@
+      let config = this.controller.recognizer.config;
+
+      let event: MouseEvent;
+      let eventDict: MouseEventInit = {
+        bubbles: true,
+        buttons: state == 'end' ? 0 : 1,
+        ...this.getSampleClientPos(sample)
+      };
+
+      switch(state) {
+        case 'start':
+          event = new MouseEvent('mousedown', eventDict);
+          break;
+        case 'move':
+          event = new MouseEvent('mousemove', eventDict);
+          break;
+        case 'end':
+          event = new MouseEvent('mouseup', eventDict);
+          break;
+      }
+
+      targetElement = targetElement || config.targetRoot;
+
+      // Implies that `targetElement` is the target element; .target cannot be set separately
+      // for mouse events, so we can't use config.mouseEventRoot.  Don't worry, the
+      // listener will still receive the event b/c bubbling.
+      targetElement.dispatchEvent(event);
+    }
+
+    /**
+     * Given a previously-recorded sequence of sampled input coordinates, this function
+     * reproduces their corresponding events and runs them through this instance's linked
+     * gesture-recognizer instance.
+     *
+     * It will re-use the original fixture-layout configuration.
+     * @param sequenceTestSpec A previously-recorded sequence of sampled input coordinates.
+     * @param replayExecutor A middleman function to use when simulating the actual events.
+     * @returns The final sample to be simulated.
+     */
+    private replayCore(sequenceTestSpec: RecordedCoordSequenceSet,
+      replayExecutor: (func: () => void, sample?: com.keyman.osk.InputSample) => void): com.keyman.osk.InputSample {
+      let inputs = sequenceTestSpec.inputs;
+      const config = sequenceTestSpec.config;
+
+      this.controller.layoutConfiguration = new FixtureLayoutConfiguration(config);
+
+      /**
+       * For each corresponding recorded sequence, notes the index of the sequence's
+       * next sample to be 'replayed'.  If the full sequence has already been
+       * replayed, its corresponding entry will be set to Number.MAX_VALUE instead.
+       */
+      let sequenceProgress: number[] = new Array(inputs.length).fill(0);
+      let sequenceTouches: Touch[] = new Array(inputs.length).fill(null);
+
+      let lastSample = null;
+
+      while(sequenceProgress.find((number) => number != Number.MAX_VALUE) !== undefined) {
+        // Determine the sequence that has the chronologically next-in-line sample
+        // to reproduce.
+        let minTimestamp = Number.MAX_VALUE;
+
+        /**
+         * Pulls double-duty as the index of the sequence owning said sample & as
+         * the sequence's common 'identifier' value for emulated Touches.
+         */
+        let selectedSequence = -1;
+
+        for(let index=0; index < inputs.length; index++) {
+          // TODO:  does not iterate over all touchpoints.  Not that we can have more than one at present...
```

Comment:
Is this supposed to be addressed in this release? If so, you should probably pull it into an issue.

Reply:
It's supposed to be addressed before the feature branch is merged, if that's what you mean.

Reply:
Just don't want it to be lost.

Reply:
Tracked as #7067.